crates/uv/src/commands/tool/install.rs

```diff
@@ -173,23 +145,45 @@ pub(crate) async fn install(
         requirements
     };
 
+    let name = from.name.to_string();
+    let installed_tools = InstalledTools::from_settings()?;
+    let existing_tool_receipt = installed_tools.get_tool_receipt(&name)?;
+
+    // If the requested and receipt requirements are the same...
+    if let Some(tool_receipt) = existing_tool_receipt.as_ref() {
+        let receipt = tool_receipt
+            .requirements()
+            .iter()
+            .cloned()
+            .map(Requirement::from)
+            .collect::<Vec<_>>();
+        if requirements == receipt {
+            // And the user didn't request a reinstall or upgrade...
+            if !force && settings.reinstall.is_none() && settings.upgrade.is_none() {
+                // We're done.
+                writeln!(printer.stderr(), "Tool `{from}` is already installed")?;
+                return Ok(ExitStatus::Failure);
+            }
+        }
+    }
+
+    // Replace entrypoints if the tool already exists (and we made it this far). If we find existing
+    // entrypoints later on, and the tool _doesn't_ exist, we'll avoid removing the external tool's
+    // entrypoints (without `--force`).
+    let reinstall_entry_points = existing_tool_receipt.is_some();
+
     // TODO(zanieb): Build the environment in the cache directory then copy into the tool directory
     // This lets us confirm the environment is valid before removing an existing install
-    let environment = installed_tools.environment(
-        &name,
-        // Do not remove the existing environment if we're reinstalling a subset of packages
-        !matches!(settings.reinstall, Reinstall::Packages(_)),
```

Comment:
It seems ok to me to only remove this when `--force` is provided?

Reply:
In other words: can't we defer to our underlying virtualenv update machinery to manage this by reinstalling packages as required?

Reply:
That seems okay, yeah.